waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/PrefixBasedDatabaseMappingService.java

```diff
@@ -96,8 +96,8 @@ private void add(AbstractMetaStore federatedMetaStore) {
       Whitelist mappedDbWhitelist = null;
       if (FederatedMetaStore.class.isAssignableFrom(federatedMetaStore.getClass())) {
         mappedDbWhitelist = getWhitelistedDatabases((FederatedMetaStore) federatedMetaStore);
+        mappedDbByPrefix.put(metaStoreMapping.getDatabasePrefix(), mappedDbWhitelist);
```

Comment:
You're 100% sure that we never need an entry in the map for a prefix against a null whitelist? By moving that here this behaviour has now changed which might be fine, I'm just checking there's definitely a reason for this.

Reply:
If we do, then it may be a problem because if `mappedDbWhitelist` is null, then the `put` method throws a NullPointerException. After testing it, I found out that it uses a ConcurrentHashMap which returns that exception if the key or value is null.

Reply:
Oh OK, that makes sense to have it like this then and if all the other tests pass then we should be fine.